kge/util/dump.py

```diff
@@ -113,83 +113,167 @@ def _add_dump_trace_parser(subparsers_dump):
+        action="store_const",
+        const=True,
+        default=False,
+        help=(
+            "Include entries from evaluation on the test data split (Note:"
+            " if none of --train --valid --test is set, this will activate"
+            " all of them jointly)."
+        ),
+    )
+    parser_dump_trace.add_argument(
+        "--search",
+        action="store_const",
+        const=True,
+        default=False,
+        help="Dump the tracefile from a search job. If a tracefile from search jobs is"
+        " used, the options --train, --valid and --test are not applicable.",
+    )
+    parser_dump_trace.add_argument(
+        "--keysfile",
+        default=False,
+        help=(
+            "A path to a file which contains lines in the format"
+            " 'new_key_name'='key_name'. For every line in the keys file, the command"
+            " will look for the value of 'key_name' in the trace entries and config and"
+            " add a respective column in the csv file with name 'new_key_name'."
+            " Additionally, for 'key_name' the special keys '$folder', '$machine'"
+            " '$checkpoint' and '$base_model' can be used. "
+        ),
+    )
+    parser_dump_trace.add_argument(
+        "--keys",
+        "-k",
+        nargs="*",
+        type=str,
+        help=(
+            "A list of 'key' entries (separated by space). For every entry,"
+            " the corresponding values of 'key' will be searched in the config and"
+            " trace entries and in the csv file a column with name 'key' will be added."
+            " The same special keys as for --keysfile can be used but must be"
+            " surrounded by quotations ''. "
+        ),
+    )
     parser_dump_trace.add_argument(
         "--checkpoint",
         default=False,
         action="store_const",
         const=True,
         help=(
-            "If source is a path to a job folder and --checkpoint is set the best "
-            "(if present) or last checkpoint will be used to determine the job_id"
+            "If source is a path to a job folder and --checkpoint is set, the best"
+            " (if present) or last checkpoint will be used to determine the job_id from"
+            " where the tracefile will be processed backwards."
         ),
     )
-
     parser_dump_trace.add_argument(
         "--job_id",
         default=False,
         help=(
-            "Specifies the training job id in the trace "
-            "from where to start processing backward"
+            "Specifies the training job id in the trace"
+            " from where to start processing the tracefile backward."
         ),
     )
-
     parser_dump_trace.add_argument(
         "--max_epoch",
         default=False,
         help=(
-            "Specifies the epoch in the trace"
-            "from where to start processing backwards"
+            "Specifies the max epoch number in the tracefile"
+            " from where to start processing backwards."
```

Comment:
Unclear. How does this differ from truncate?

Reply:
Truncate only works together with a checkpoint. When a checkpoint is provided then --truncate will enforce the max_epoch number will be the one the checkpoint holds. On the other hand, --max_epoch works in any case and just sets the max_epoch number. Using both together does not make sense and I added a check to enforce this.